spec/requests/forms/what_happens_next_controller_spec.rb

```diff
@@ -104,5 +108,107 @@
     it "Redirects you to the form overview page" do
       expect(response).to redirect_to(form_path(2))
     end
+
+    context "when previewing markdown" do
+      let(:route_to) { "preview" }
+      let(:what_happens_next_text) { nil }
+      let(:what_happens_next_markdown) { "[a link](https://example.com)" }
+
+      it "reads the existing form" do
+        expect(form).to have_been_read
+      end
+
+      it "renders the what happens next template" do
+        expect(response).to have_rendered("forms/what_happens_next/new")
+      end
+
+      it "returns 200" do
+        expect(response).to have_http_status(:ok)
+      end
+
+      it "renders the guidance markdown as html" do
+        expect(response.body).to include('<a href="https://example.com" class="govuk-link" rel="noreferrer noopener" target="_blank">a link</a>')
+      end
+
+      context "when markdown is invalid" do
+        let(:what_happens_next_markdown) { "# A level one heading" }
+
+        it "reads the existing form" do
+          expect(form).to have_been_read
+        end
+
+        it "renders the template" do
+          expect(response).to have_rendered("forms/what_happens_next/new")
+        end
+
+        it "returns 200" do
+          expect(response).to have_http_status(:ok)
+        end
+      end
+    end
+
+    context "when saving markdown" do
+      let(:route_to) { "save_and_continue" }
+
+      it "reads the existing form" do
+        expect(form).to have_been_read
+      end
+
+      it "redirects the user to the form overview page" do
+        expect(response).to redirect_to(form_path(2))
+      end
+
+      context "when markdown is invalid" do
+        let(:what_happens_next_markdown) { "# A level one heading" }
+
+        it "returns 422" do
+          expect(response).to have_http_status(:unprocessable_entity)
+        end
+
+        it "renders the template" do
+          expect(response).to have_rendered("forms/what_happens_next/new")
+        end
+      end
+    end
+  end
+
+  describe "#render_preview" do
+    let(:markdown) { "- Markdown" }
+
+    before do
+      post what_happens_next_render_preview_path(form_id: form.id), params: { markdown: }
+    end
+
+    it "returns a JSON object containing the converted HTML" do
+      expect(response.body).to eq({ preview_html: "<ul class=\"govuk-list govuk-list--bullet\">\n  <li>Markdown</li>\n\n</ul>", errors: [] }.to_json)
+    end
+
+    it "returns 200" do
+      expect(response).to have_http_status(:ok)
+    end
+
+    context "when markdown is blank" do
+      let(:markdown) { "" }
+
+      it "returns a JSON object containing the converted HTML" do
+        expect(response.body).to eq({ preview_html: I18n.t("guidance.no_guidance_added_html"), errors: [] }.to_json)
+      end
+
+      it "returns 200" do
+        expect(response).to have_http_status(:ok)
```

Comment:
Should this be 422?

Reply:
No - blank markdown is valid in this case (blank text was valid before)